cpp/src/arrow/memory_pool.cc

```diff
@@ -916,6 +916,13 @@ class PoolBuffer final : public ResizableBuffer {
   }
 
  private:
+  static Result<int64_t> RoundCapacity(int64_t capacity) {
+    if (capacity > std::numeric_limits<int64_t>::max() - 63) {
```

Comment:
It's a bit late in my tz, so I maybe stupid. why not `std::numeric_limits<int64_t>::max() - 64`? Since

```
// Returns 'value' rounded up to the nearest multiple of 'factor' when factor
// is a power of two.
// The result is undefined on overflow, i.e. if `value > 2**64 - factor`,
// since we cannot return the correct result which would be 2**64.
constexpr int64_t RoundUpToPowerOf2(int64_t value, int64_t factor) {
  // DCHECK(value >= 0);
  // DCHECK(IsPowerOf2(factor));
  return (value + (factor - 1)) & ~(factor - 1);
}
```

Reply:
(So the comment in `RoundUpToPowerOf2` can be `value > 2**64 - factor + 1` ? Since `factor` is it's argument )

Reply:
No, `std::numeric_limits<int64_t>::max()` is `2**63 - 1`, not `2**63`.

Reply:
Oh, my bad!